src/crimson/osd/heartbeat.cc

```diff
@@ -309,6 +312,29 @@ seastar::future<> Heartbeat::handle_you_died()
   return seastar::now();
 }
 
+void Heartbeat::heartbeat_check()
+{
+  bool failure = false;
+  for (auto& p: peers) {
+    if (clock::is_zero(p.second.first_tx))
+      continue;
+
+    if (p.second.is_unhealthy(clock::now())) {
+      std::time_t deadline = clock::to_time_t(
```

Comment:
instead of defining a local variable for printing, i'm inclined to reuse the formatter. i extract the one used by messenger in #33849 . could you give it a try?

Reply:
Ok, I'll try it:-)

Reply:
@xxhdx1985126 this is not addressed.